engine/timer_task_conrrent_processor.go

```diff
@@ -113,14 +114,35 @@ func (w *timerTaskConcurrentProcessor) processTimerTask(
 	case persistence.TimerTaskTypeWorkerTaskBackoff:
 		return w.processTimerTaskWorkerTaskBackoff(task)
 	case persistence.TimerTaskTypeProcessTimeout:
-		panic("TODO")
+		return w.processTimerTaskProcessTimeout(task)
 	case persistence.TimerTaskTypeTimerCommand:
 		return w.processTimerTaskForTimerCommand(task)
 	default:
 		panic(fmt.Sprintf("unknown timer task type %v", task.TaskType))
 	}
 }
 
+func (w *timerTaskConcurrentProcessor) processTimerTaskProcessTimeout(
+	task persistence.TimerTask,
+) error {
+	resp, err := w.store.ProcessTimerTaskForProcessTimeout(w.rootCtx, persistence.ProcessTimerTaskRequest{
+		Task: task,
+	})
+	if err != nil {
+		return err
+	}
+
+	if resp == nil {
```

Comment:
In Golang, usually we don't check this. `err != nil` has indicated that this field is not nil, otherwise the next line will panic as well as the same results.